modules/mobile/app/models/mobile/v0/authorized_services.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Mobile
+  module V0
+    class AuthorizedServices
+      def initialize(user)
+        @user = user
+      end
+
+      def to_a
+        authorized_services_map.select { |_, v| v == true }.keys
+      end
+
+      def available_services
+        authorized_services_map.keys
+      end
+
+      private
+
+      def authorized_services_map
+        {
+          appeals: access?(appeals: :access?),
+          appointments: true,
+          claims: flagged_access?(:mobile_lighthouse_claims, { lighthouse: :access? }, { evss: :access? }),
+          directDepositBenefits: flagged_access?(:mobile_lighthouse_direct_deposit, { lighthouse: :mobile_access? },
+                                                 { evss: :access?, ppiu: :access? }),
+          directDepositBenefitsUpdate: direct_deposit_update_access?,
+          disabilityRating: flagged_access?(:mobile_lighthouse_disability_ratings, { lighthouse: :access? },
+                                            { evss: :access? }),
+          lettersAndDocuments: flagged_access?(:mobile_lighthouse_letters, { lighthouse: :access? },
+                                               { evss: :access? }),
+          militaryServiceHistory: access?(vet360: :military_access?),
+          paymentHistory: access?(bgs: :access?),
+          userProfileUpdate: access?(vet360: :access?),
+          secureMessaging: access?(mhv_messaging: :access?),
+          scheduleAppointments: access?(schedule_appointment: :access?),
+          prescriptions: access?(mhv_prescriptions: :access?),
+          preferredName: access?(demographics: :access_update?) && access?(mpi: :queryable?),
+          genderIdentity: access?(demographics: :access_update?) && access?(mpi: :queryable?)
+        }
+      end
+
+      def flagged_access?(flag_name, flag_on_policy, flag_off_policy)
+        if Flipper.enabled?(flag_name, @user)
+          access?(flag_on_policy)
+        else
+          access?(flag_off_policy)
+        end
+      end
+
+      def access?(policies)
+        policies.all? do |policy_name, policy_rule|
+          if policy_rule.is_a? Array
+            policy_rule.all? { |rule| access?({ policy_name => rule }) }
+          else
+            @user.authorize(policy_name, policy_rule)
+          end
+        end
+      end
+
```

Comment:
I'm surprised that we don't have to do a similar PPIU check for lighthouse, given that evss and lighthouse use the same upstream servers and should have the same access rules. But the ppiu policy is specific to evss. My suspicion is that the lighthouse direct deposit team just never created a check for this and will probably cause some users to attempt to update the direct deposit access and experience errors.

Reply:
Make a note of this in upcoming lighthouse policy check ticket. 6016